File: src/main/java/com/github/johanfredin/bindygenerator/GeneratorConfig.java
package com.github.johanfredin.bindygenerator;

public class GeneratorConfig {

    private final String delimiter;
    private final String javaClassName;
    private final FieldMapping fieldMapping;
    private final boolean isHeader;
    private final boolean useNumericFieldTypes;
    private final boolean usePrimitiveTypesWherePossible;
    private final boolean includeColumnName;


    public GeneratorConfig(String delimiter, String javaClassName,
                           FieldMapping fieldMapping, boolean isHeader,
                           boolean useNumericFieldTypes, boolean usePrimitiveTypesWherePossible,
                           boolean includeColumnName) {
        this.delimiter = delimiter;
        this.javaClassName = javaClassName;
        this.fieldMapping = fieldMapping;
        this.isHeader = isHeader;
        this.useNumericFieldTypes = useNumericFieldTypes;
        this.usePrimitiveTypesWherePossible = usePrimitiveTypesWherePossible;
        this.includeColumnName = includeColumnName;
    }

    public String getDelimiter() {
        return delimiter;
    }

    public boolean isHeader() {
        return isHeader;
    }

    public boolean isUseNumericFieldTypes() {
        return useNumericFieldTypes;
    }

    public String getJavaClassName() {
        return javaClassName;
    }

    public FieldMapping getFieldMapping() {
        return fieldMapping;
    }

    public boolean isUsePrimitiveTypesWherePossible() {
        return usePrimitiveTypesWherePossible;
    }

    public boolean isIncludeColumnName() {
        return includeColumnName;
    }

    @Override
    public String toString() {
        return "GeneratorConfig{" +
                "delimiter='" + delimiter + '\'' +
                ", javaClassName='" + javaClassName + '\'' +
                ", fieldMapping=" + fieldMapping +
                ", isHeader=" + isHeader +
                ", useNumericFieldTypes=" + useNumericFieldTypes +
                ", usePrimitiveTypesWherePossible=" + usePrimitiveTypesWherePossible +
                ", includeColumnName=" + includeColumnName +
                '}';
    }
}

File: src/main/java/com/github/johanfredin/bindygenerator/GeneratorConfig.kt
package com.github.johanfredin.bindygenerator

/**
 * The configuration of the generator
 * @property delimiter the field delimiter used in the datasource file
 * @property javaClassName the name of the java class that will represent the datasource file
 * @property fieldMapping what mapping convention to use for the field names when turning them into java fields
 * @property isHeader whether or not a header record is present (typically the first row in the file)
 * @property integerType the field type to use for integers (defaults to string)
 * @property decimalType the field type to use for decimals (defaults to string)
 * @property stringType the field type to use for strings (defaults to string)
 * @property isIncludeColumnName whether or not to include the columnName argument into the bindy &#64;DataField annotation
 *
 */
data class GeneratorConfig(val delimiter: String,
                           val javaClassName: String,
                           val fieldMapping: FieldMapping,
                           val isHeader: Boolean,
                           val isIncludeColumnName: Boolean,
                           val integerType: IntegerType,
                           val decimalType: DecimalType,
                           val stringType: `StringType.java`)

